### DISCUSSION TITLE
Should private methods be ignored within a request object?

### DISCUSSION THREAD

**Discussion by cristalp** [smallrye/smallrye-open-api#1964] [General]
I have an object that's used as request body, and it looks something like this:

```java
@Schema(description = "Bla bla")
public class ParameterObject {

  @Schema(required = false, description = "The date")
  private LocalDate date;

  @AssertTrue(message = "The date is invalid")
  private boolean isDateValid() {
     // Some validation
  }

  public LocalDate getDate() {
    return this.date;
  }

  public void setDate(final LocalDate date) {
    this.date = date;
  }
}
```

The method `isDateValid` is only used for Bean Validation and is not part of the object's "API". So I would expect to see the following as example in SwaggerUI:

```json
{
  "date": "2022-03-10"
}
```

However, I see this:
```json
{
  "date": "2022-03-10",
  "dateValid": true
}
```

I have to annotate the method with `@JsonIgnore` to remove it from the example.

But I think that private methods should not be scanned at all, even if they stick to the bean method standard. Or am I wrong?

**Comment by MikeEdgar** [comment]
@cristalp , private properties are enabled by default, but you can disable by using
```
mp.openapi.extensions.smallrye.private-properties.enable=false
```

**Reply by cristalp** [reply]
Thanks! I'm not quite sure, though, how to configure this. Can I configure it directly on the Maven plugin? Or do I need a separate configuration file and reference it with `configProperties`?

**Reply by MikeEdgar** [reply]
Ah, I guess that one wasn't added to the plugin yet. You can use this in the configuration instead:
```xml
<systemPropertyVariables>
  <mp.openapi.extensions.smallrye.private-properties.enable>false</mp.openapi.extensions.smallrye.private-properties.enable>
</systemPropertyVariables>
```

**Reply by cristalp** [reply]
Thanks, that worked like a charm!
